src/editions.md

```diff
@@ -0,0 +1,39 @@
+# Editions
+
+Before we get into the details of the language itself, we need to cover the
+concept of an "edition."
+
+An "edition" of Rust consists of a particular year. There are two editions of
+Rust, currently:
+
+* Rust 2015
+* Rust 2018
+
+Conforming compilers are expected to:
+
+* Provide a flag allowing users to select an edition.
+* Default to the 2015 edition if no edition is selected.
+
+The reference follows this guideline; everything stated in this document is
+part of the 2015 edition unless specifically marked otherwise.
+
+## What editions can change
```

Comment:
This feels like the core definition of an edition, and IMO, should be before the list of years. We're also missing that it's a per-crate setting. And we want to talk about them today, and not just new ones. Perhaps something like (if not exactly)

```
*Editions* are a per-crate setting that can only do the following two things:

* Change a deprecation from the prior edition into a hard error
* Change a deprecation from the prior edition to deny by default, and ...
```

---

Hmm, I just realized that this means that the deprecatedness of a feature and in which edition it was deprecated in are now part of the language proper (and in-scope for the reference) and not just a part of linting (which is currently out-of-scope for the reference)